internal/internal_task_handlers.go

```diff
@@ -703,13 +709,13 @@ func (wth *workflowTaskHandlerImpl) ProcessWorkflowTask(
 			tagPreviousStartedEventID, task.GetPreviousStartedEventId())
 	})
 
-	workflowContext, err := wth.getOrCreateWorkflowContext(task, workflowTask.historyIterator)
+	contextCached, workflowContext, err := wth.getOrCreateWorkflowContext(task, workflowTask.historyIterator)
```

Comment:
can we put isCached into the workflowContext itself so do don't need to know it here.

Reply:
I think it's only needed for this area, I am hesitant to add yet more values into the context (values in context are basically untyped dynamic fields, whereas this param explicitly lets the reader/caller know that it is required to move through these calls).

Reply:
I mean a strong typed field in workflowExecutionContextImpl.

Reply:
I see. The field would still have no value except as a return here and a param to the Unlock. There is no other use of the field, so it's seems strange to add it there. Is there a good reason to add it even though its only use will be a few lines below?

Reply:
I think that is some implementation detail that can be hide away and caller does not need to know about it.